lib/statistex.ex

```diff
@@ -541,10 +595,85 @@ defmodule Statistex do
 
   def median(samples, options) do
     percentiles =
-      Keyword.get_lazy(options, :percentiles, fn -> percentiles(samples, @median_percentile) end)
+      Keyword.get_lazy(options, :percentiles, fn ->
+        Percentile.percentiles(samples, @median_percentile)
+      end)
+
+    get_percentile(samples, @median_percentile, percentiles)
+  end
+
+  @doc """
+  Calculates the lower and upper bound for outliers.
+
+  Any sample that is `<` as the lower bound and any sample `>` are outliers of
+  the given `samples`.
+
+  ## Examples
+
+      iex> Statistex.outliers_bounds([3, 4, 5])
+      {3, 5}
+
+      iex> Statistex.outliers_bounds([1, 2, 6, 50, 50, 50, 50, 50, 50, 50, 50, 50, 50, 50])
+      {22.5, 50}
+
+      iex> Statistex.outliers_bounds([50, 50, 50, 50, 50, 50, 50, 50, 50, 50, 50, 99, 99, 99])
+      {50, 80.625}
+  """
+  @spec outliers_bounds(samples, keyword) :: {lower :: number, upper :: number}
+  def outliers_bounds(samples, options \\ [])
+  def outliers_bounds([], _), do: raise(ArgumentError, @empty_list_error_message)
+  def outliers_bounds(samples, options), do: samples |> Enum.sort() |> do_outliers_bounds(options)
+
+  defp do_outliers_bounds(samples, options) do
+    percentiles =
+      Keyword.get_lazy(options, :percentiles, fn -> Percentile.percentiles(samples, [25, 75]) end)
+
+    minimum = Keyword.get_lazy(options, :minimum, fn -> hd(samples) end)
+    maximum = Keyword.get_lazy(options, :maximum, fn -> List.last(samples) end)
+
+    p25 = get_percentile(samples, 25, percentiles)
+    p75 = get_percentile(samples, 75, percentiles)
+    iqr = p75 - p25
+
+    {max(p25 - iqr * @iqr_factor, minimum), min(p75 + iqr * @iqr_factor, maximum)}
+  end
+
+  @doc """
+  Returns all outliers for the given `samples`.
+
+  ## Examples
+
+      iex> Statistex.outliers([3, 4, 5])
+      []
+
+      iex> Statistex.outliers([1, 2, 6, 50, 50, 50, 50, 50, 50, 50, 50, 50, 50, 50])
+      [1, 2, 6]
+
+      iex> Statistex.outliers([50, 50, 50, 50, 50, 50, 50, 50, 50, 50, 50, 99, 99, 99])
+      [99, 99, 99]
+  """
+  @spec outliers(samples, keyword) :: samples | []
+  def outliers(samples, options \\ []) do
+    {outliers, _rest} = samples |> Enum.sort() |> do_outliers(options)
+
+    outliers
+  end
+
+  defp do_outliers(samples, options) do
+    {lower_bound, upper_bound} =
+      Keyword.get_lazy(options, :outliers_bounds, fn -> do_outliers_bounds(samples, options) end)
+
+    {min, rest} = Enum.split_while(samples, fn sample -> sample < lower_bound end)
+
+    {max, rest} =
+      rest |> Enum.reverse() |> Enum.split_while(fn sample -> sample > upper_bound end)
+
+    {min ++ max, rest}
```

Comment:
I think naming them `_outliers` (probably `low_outliers`/`high_outliers`) would be a bit clearer, `min` and `max` I thought about the minimum and maximum values we also compute.